https://issues.redhat.com/browse/ACM-13415 Unknown state troubleshoot (#6936)
open-cluster-management/rhacm-docs: clusters/support_troubleshooting/trouble_unknown_rosa_hcp.adoc

[#trouble-state-unknown-rosa-hcp]

> = Troubleshooting all managed clusters become `Unknown` on a ROSA with HCP cluster

Reviewer: Same comment, HCP should be written out to hosted control planes

> On a Red Hat OpenShift Service on AWS (ROSA) with hosted control planes (HCP) cluster, the state of all managed clusters might switch to `Unknown`.

Reviewer: ```suggestion
On a Red Hat OpenShift Service on AWS (ROSA) with hosted control planes cluster, the state of all managed clusters might switch to `Unknown`.
```

[#symptom-state-unknown-rosa-hcp]
== Symptom: All managed clusters become `Unknown` on a ROSA with HCP cluster

> The state of all managed cluster on a {rosa} hosted cluster suddenly becomes `Unknown`. When you check the `klusterlet-agent` pod log in the `open-cluster-management-agent` namespace on your managed cluster, you see an error that resembles the following: 

Reviewer: ```suggestion
The state of all managed cluster on a {rosa} hosted cluster suddenly becomes `Unknown`. When you check the `klusterlet-agent` pod log in the `open-cluster-management-agent` namespace on your managed cluster, you see the following error message: 
```

> The state of all managed cluster on a {rosa} hosted cluster suddenly becomes `Unknown`. When you check the `klusterlet-agent` pod log in the `open-cluster-management-agent` namespace on your managed cluster, you see the following error message: 

Reviewer: The {rosa} conref needs to be removed from this line because we don't have a conref created for 2.10.

https://github.com/stolostron/rhacm-docs/blob/2.10_stage/modules/common-attributes.adoc

Author: Wow where's the melting face emote when you need it, thanks for catching that

Reviewer: No worries!

[source,bash]
----
E0809 18:45:29.450874       1 reflector.go:147] k8s.io/client-go@v0.29.4/tools/cache/reflector.go:229: Failed to watch *v1.CertificateSigningRequest: failed to list *v1.CertificateSigningRequest: Get "https://api.xxx.openshiftapps.com:443/apis/certificates.k8s.io/v1/certificatesigningrequests?limit=500&resourceVersion=0": tls: failed to verify certificate: x509: certificate signed by unknown authority
----

[#resolve-state-unknown-rosa-hcp]
== Resolving the problem: All managed clusters become `Unknown` on a ROSA with HCP cluster

To resolve the problem, add the `ISRG Root X1` certificate to the klusterlet CA bundle by completing the following steps:

. Download and encode the root CA certificate by running the following command:

+
[source,bash]
----
curl -s https://letsencrypt.org/certs/isrgrootx1.pem | base64 | tr -d "\n"
----

> . Create a `KlusterletConfig` resource and add your encoded CA certificate to the `spec`. See the following example:

Reviewer: ```suggestion
. Create a `KlusterletConfig` resource and add your encoded CA certificate to the `spec` parameter section. See the following example:
```

+
[source,yaml]
----
apiVersion: config.open-cluster-management.io/v1alpha1
kind: KlusterletConfig
metadata:
  name: isrg-root-x1-ca
spec:
  hubKubeAPIServerCABundle: "<your_ca_certificate>"
----

. Apply the resource by running the following command on the hub cluster:

+
[source,bash]
----
oc apply -f <filename>
----

. Annotate your managed cluster by running the following command on the hub cluster. Replace values where needed:

+
[source,bash]
----
oc annotate --overwrite managedcluster <cluster_name> agent.open-cluster-management.io/klusterlet-config='isrg-root-x1-ca'
----
+
The state of some managed clusters might recover.

. For managed clusters that remain `Unknown`, export and decode the `import.yaml` file from the hub cluster by running the following command on the hub cluster. Replace values where needed:

+
[source,bash]
----

> oc get secret <cluster_name>-import -n <cluster_name> -o jsonpath={.data.import .yaml} | base64 --decode > <cluster_name>-import.yaml

Reviewer 2: Use `jsonpath={.data.import\.yaml}` instead.

Author: Updated

[source,bash]
----
E0809 18:45:29.450874       1 reflector.go:147] k8s.io/client-go@v0.29.4/tools/cache/reflector.go:229: Failed to watch *v1.CertificateSigningRequest: failed to list *v1.CertificateSigningRequest: Get "https://api.xxx.openshiftapps.com:443/apis/certificates.k8s.io/v1/certificatesigningrequests?limit=500&resourceVersion=0": tls: failed to verify certificate: x509: certificate signed by unknown authority
----

[#resolve-state-unknown-rosa-hcp]
== Resolving the problem: All managed clusters become `Unknown` on a {rosa} hosted cluster

To resolve the problem, add the `ISRG Root X1` certificate to the klusterlet CA bundle by completing the following steps:

. Download and encode the root CA certificate by running the following command:

+
[source,bash]
----
curl -s https://letsencrypt.org/certs/isrgrootx1.pem | base64 | tr -d "\n"
----

. Create a `KlusterletConfig` resource and add your encoded CA certificate to the `spec`. See the following example:

+
[source,yaml]
----
apiVersion: config.open-cluster-management.io/v1alpha1
kind: KlusterletConfig
metadata:
  name: isrg-root-x1-ca
spec:
  hubKubeAPIServerCABundle: "<your_ca_certificate>"
----

> . Apply the resource by running the following command:

Reviewer 2: . Apply the resource by running the following command on the hub cluster:

Author: Updated

+
[source,bash]
----
oc apply -f <filename>
----

> . Annotate your managed cluster by running the following command. Replace values where needed:

Reviewer 2: . Annotate your managed cluster by running the following command on the hub cluster. Replace values where needed:

Author: Updated

+
[source,bash]
----

> oc annotate --overwrite <cluster_name> local-cluster agent.open-cluster-management.io/klusterlet-config='isrg-root-x1-ca'

Reviewer 2: ```
oc annotate --overwrite managedcluster <cluster_name> agent.open-cluster-management.io/klusterlet-config='isrg-root-x1-ca'
```

Author: Updated

----
+
The state of some managed clusters might recover.

> . For managed clusters that remain `Unknown`, export the `import.yaml` file from the hub cluster by running the following command. Replace values where needed:

Reviewer 2: . For managed clusters that remain `Unknown`, export the `import.yaml` file from the hub cluster by running the following command on the hub cluster. Replace values where needed:
Comment

Author: Updated

+
[source,bash]
----
oc get secret <cluster_name>-import -n <cluster_name> -o jsonpath={.data.import .yaml}
----

. Decode the file by running the following command. Replace values where needed:

+
[source,bash]
----
base64 --decode > <cluster_name>-import.yaml

> ----

Reviewer 2: Replace with command below
```
oc get secret <cluster_name>-import -n <cluster_name> -o jsonpath='{.data.import\.yaml}' | base64 --decode > <cluster_name>-import.yaml
```

Author: Same thing here, we generally want to separate commands but I can ask the team and get some feedback

Reviewer 2: Similarly, if you believe it's necessary to separate them, please refer to the content below:

---

5. For managed clusters that remain Unknown, export the import.yaml file from the hub cluster by running the following command on the hub cluster. Replace values where needed:
```
oc get secret <cluster_name>-import -n <cluster_name> -o jsonpath='{.data.import\.yaml}' > <cluster_name>-import-encoded.yaml
```
6. Decode the file by running the following command. Replace values where needed:
```
base64 --decode > <cluster_name>-import-encoded.yaml > <cluster_name>-import.yaml
```
7. ...

Author: I asked the team again but generally it is required that we separate commands. I'll get back to you on this

Author: We can combine it

> . Apply the file by running the following command. Replace values where needed:

Reviewer 2: . Apply the file by running the following command on the managed cluster. Replace values where needed:

Author: Updated

[#trouble-state-unknown-rosa-hcp]
= Troubleshooting all managed clusters become `Unknown` on a ROSA with HCP cluster

On a Red Hat OpenShift Service on AWS (ROSA) with hosted control planes (HCP) cluster, the state of all managed clusters might switch to `Unknown`.

[#symptom-state-unknown-rosa-hcp]

> == Symptom: All managed clusters become `Unknown` on a {rosa} hosted cluster

Reviewer 3: ```suggestion
== Symptom: All managed clusters become `Unknown` on a ROSA with HCP cluster
```

Reviewer 3: I don't seem to be able to comment on lines 9 and 17, but I'd replace `{rosa} hosted cluster` with `ROSA with HCP cluster`.
